kpatch-build/kpatch-build

```diff
@@ -610,14 +640,36 @@ if [[ ! -z $UBUNTU_KERNEL ]]; then
 fi
 
 cd "$TEMPDIR/output"
-ld -r -o ../patch/output.o $(find . -name "*.o") >> "$LOGFILE" 2>&1 || die
-md5sum ../patch/output.o | awk '{printf "%s\0", $1}' > checksum.tmp || die
-objcopy --add-section .kpatch.checksum=checksum.tmp --set-section-flags .kpatch.checksum=alloc,load,contents,readonly  ../patch/output.o || die
-rm -f checksum.tmp
+ld -r $KPATCH_LDFLAGS -o ../patch/tmp_output.o $(find . -name "*.o") >> "$LOGFILE" 2>&1 || die
+
+if $KPATCH_MODULE; then
+	# Add .kpatch.checksum for kpatch script
+	md5sum ../patch/tmp_output.o | awk '{printf "%s\0", $1}' > checksum.tmp || die
+	objcopy --add-section .kpatch.checksum=checksum.tmp --set-section-flags .kpatch.checksum=alloc,load,contents,readonly  ../patch/tmp_output.o || die
+	rm -f checksum.tmp
+	"$TOOLSDIR"/create-kpatch-module $TEMPDIR/patch/tmp_output.o $TEMPDIR/patch/output.o 2>&1 |tee -a "$LOGFILE"
+	check_pipe_status create-kpatch-module
+else
+	cp $TEMPDIR/patch/tmp_output.o $TEMPDIR/patch/output.o || die
+
+ fi
+
 cd "$TEMPDIR/patch"
-KPATCH_BUILD="$SRCDIR" KPATCH_NAME="$PATCHNAME" KBUILD_EXTRA_SYMBOLS="$KBUILD_EXTRA_SYMBOLS" \
+
+KPATCH_BUILD="$SRCDIR" KPATCH_NAME="$PATCHNAME" \
+KBUILD_EXTRA_SYMBOLS="$KBUILD_EXTRA_SYMBOLS" \
+KPATCH_LDFLAGS="$KPATCH_LDFLAGS" \
 	make "O=$OBJDIR" >> "$LOGFILE" 2>&1 || die
 
+if ! $KPATCH_MODULE; then
+	if [[ -z "$KPATCH_LDFLAGS" ]]; then
+		extra_flags="--no-klp-arch-sections"
+	fi
+	cp $TEMPDIR/patch/kpatch-$PATCHNAME.ko $TEMPDIR/patch/kpatch-tmp.ko || die
```

Comment:
I think this would break if the patch were named `tmp.patch`.  Maybe just call it tmp.ko?